context/app/static/js/pages/Genes/hooks.js

```diff
@@ -0,0 +1,36 @@
+import { useState, useEffect } from 'react';
+import useSWR from 'swr';
+
+async function fetchGeneCommonName(geneNamesURL) {
+  const response = await fetch(geneNamesURL, { headers: { Accept: 'application/json' } });
+
+  if (!response.ok) {
+    console.error('Genenames API failed.', response);
+  }
+
+  const results = await response.json();
+
+  return results;
+}
+
+function useGeneCommonName(geneSymbol) {
+  const [commonName, setCommonName] = useState('');
+
+  const { data: geneCommonName } = useSWR(
+    `https://rest.genenames.org/fetch/symbol/${geneSymbol}`,
+    fetchGeneCommonName,
+    {
+      fallbackData: '',
+    },
+  );
+
+  useEffect(() => {
+    if (geneCommonName && geneCommonName.response && geneCommonName.response.docs[0]) {
+      setCommonName(geneCommonName.response.docs[0].name);
+    }
+  }, [geneCommonName]);
+
+  return commonName;
```

Comment:
- This `useState` and `useEffect` are unnecessary since the set value is entirely derived from the already-stateful `useSWR` hook. This [react doc article](https://react.dev/learn/you-might-not-need-an-effect) goes into more detail, but essentially this is similar to the `firstName`/`lastName`/`fullName` example described there, where fullName is better represented as a derived const than as its own state.
- We should be able to simplify it by using [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) to drill into potentially undefined values.
- We should also hook into `useSWR`'s `isLoading` boolean to provide an accurate loading state
- The `fallbackData` provided to `useSWR` should correspond to the shape of the expected response from the fetcher function. 
- With all of the above in mind, the fetcher function could match its name more closely by extracting the common name itself, which should make the `useState` and `useEffect` hook even more unnecessary :)

```suggestion
async function fetchGeneCommonName(geneNamesURL) {
  const response = await fetch(geneNamesURL, { headers: { Accept: 'application/json' } });

  if (!response.ok) {
    console.error('Genenames API failed.', response);
  }

  const results = await response.json();
  
  const commonName = results?.response?.docs?[0]?.name;

  return commonName;
}

function useGeneCommonName(geneSymbol) {

  const { data: commonName, isLoading } = useSWR(
    `https://rest.genenames.org/fetch/symbol/${geneSymbol}`,
    fetchGeneCommonName,
    {
      fallbackData: '',
    },
  );

  return {commonName, isLoading};
```

Reply:
There's also the question of useSWR's default behavior being to revalidate the query every time the window is refocused: https://swr.vercel.app/docs/revalidation

I'm not sure if that's necessary for data this static - we could probably quiet down our API calls by adding `revalidateOnFocus: false` to the useSWR config.

Reply:
Thanks for this, learning more about swr